src/java.desktop/share/classes/sun/swing/FilePane.java

```diff
@@ -1189,19 +1194,46 @@ public Component getTableCellRendererComponent(JTable table, Object value,
                 Icon icon = chooser.getIcon(file);
                 setIcon(icon);
 
-            } else if (value instanceof Long) {
-                long len = ((Long) value) / 1024L;
+            } else if (value instanceof Long len) {
                 if (listViewWindowsStyle) {
-                    text = MessageFormat.format(kiloByteString, len + 1);
-                } else if (len < 1024L) {
-                    text = MessageFormat.format(kiloByteString, (len == 0L) ? 1L : len);
+                    updateMessageFormatPattern(kiloByteString, 1);
+                    if (len == 0) {
+                        objs[0] = Double.valueOf(0.0);
+                    } else if (len > 0 && len < 100L) {
+                        objs[0] = Double.valueOf(0.1);
+                    } else {
+                        double kbVal = formatToDoubleValue(len);
+                        objs[0] = Double.valueOf(kbVal);
+                    }
+                    text = mf.format(objs);
+                } else if (len < 100L) {
```

Comment:
we can never reach here, can we ? And isn't it obsolete ?

Reply:
It's reachable when `listViewWindowsStyle` is `false`, which seems to be the case in all the Look-and-Feels but Windows one.

Yet it looks confusing.

Reply:
> It's reachable when `listViewWindowsStyle` is `false`, which seems to be the case in all the Look-and-Feels but Windows one.
> 
> Yet it looks confusing.

We have
                if (listViewWindowsStyle) {
                     ..
                 } else if (len < 1024L) {
                     ...
                 } else if (len < 100L) {

So if listViewWindowsStyle is false then we get to < 1024 case and if there len is (say) 20
we enter that, so as far as I can see we can never reach the < 100 case.

Reply:
} else if (len < 1024L) {
...
} has been removed. 
So if listViewWindowStyle is false then the next else if condition will be executed i.e. <100L case.